vxquery-core/src/main/java/org/apache/vxquery/compiler/rewriter/rules/ConsolidateDescandantChild.java

```diff
@@ -0,0 +1,122 @@
+package org.apache.vxquery.compiler.rewriter.rules;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.lang3.mutable.Mutable;
+import org.apache.vxquery.compiler.rewriter.rules.util.ExpressionToolbox;
+import org.apache.vxquery.compiler.rewriter.rules.util.OperatorToolbox;
+import org.apache.vxquery.functions.BuiltinFunctions;
+import org.apache.vxquery.functions.BuiltinOperators;
+
+import edu.uci.ics.hyracks.algebricks.common.exceptions.AlgebricksException;
+import edu.uci.ics.hyracks.algebricks.core.algebra.base.ILogicalExpression;
+import edu.uci.ics.hyracks.algebricks.core.algebra.base.ILogicalOperator;
+import edu.uci.ics.hyracks.algebricks.core.algebra.base.ILogicalPlan;
+import edu.uci.ics.hyracks.algebricks.core.algebra.base.IOptimizationContext;
+import edu.uci.ics.hyracks.algebricks.core.algebra.base.LogicalExpressionTag;
+import edu.uci.ics.hyracks.algebricks.core.algebra.base.LogicalOperatorTag;
+import edu.uci.ics.hyracks.algebricks.core.algebra.base.LogicalVariable;
+import edu.uci.ics.hyracks.algebricks.core.algebra.expressions.AbstractFunctionCallExpression;
+import edu.uci.ics.hyracks.algebricks.core.algebra.expressions.VariableReferenceExpression;
+import edu.uci.ics.hyracks.algebricks.core.algebra.functions.IFunctionInfo;
+import edu.uci.ics.hyracks.algebricks.core.algebra.operators.logical.AbstractUnnestOperator;
+import edu.uci.ics.hyracks.algebricks.core.rewriter.base.IAlgebraicRewriteRule;
+
+/**
+ * The rule searches for assign operator with an aggregate function expression
+ * immediately following an aggregate operator with a sequence expression.
+ * XQuery aggregate functions are implemented in both scalar (one XDM Instance
+ * input as a sequence) and iterative (a stream of XDM Instances each one is
+ * single object).
+ * 
+ * <pre>
+ * Before 
+ * 
+ *   plan__parent
+ *   UNNEST( $v2 : fn:child( args[$v1] ) )
+ *   UNNEST( $v1 : fn:descandant( args[$v0] ) )
+ *   plan__child
+ *   
+ *   
+ * After
+ * 
+ *   plan__parent
+ *   UNNEST( $v2 : fn:descandant( args[$v0] ) )
+ *   plan__child
+ * 
+ * 
+ * </pre>
+ * 
+ * @author shivanimall
+ */
+
+public class ConsolidateDescandantChild implements IAlgebraicRewriteRule {
+
+    @Override
+    public boolean rewritePre(Mutable<ILogicalOperator> opRef, IOptimizationContext context) throws AlgebricksException {
+        // TODO Auto-generated method stub
+        return false;
+    }
+
+    @Override
+    public boolean rewritePost(Mutable<ILogicalOperator> opRef, IOptimizationContext context)
+            throws AlgebricksException {
+
+        boolean modified = false;
```

Comment:
I don't see any reason to have modified. Remove this variable and return true or false.